src/histogram_equalization.jl

```diff
@@ -77,28 +77,54 @@ Convert an array of RGB channel data to grayscale in the range [0, 255].
 
 Identical to MATLAB `rgb2gray` (https://www.mathworks.com/help/matlab/ref/rgb2gray.html).
 """
-function rgb2gray(rgbchannels::Array{Float64,3}) 
-# dmw: Can we set this up to return a Gray image instead of an int matrix?
-# dmw: Check whether the coefficients differ substantially -- we can make it an alternative to the Gray function, like GrayMLB and apply it like Gray.()
+function rgb2gray(rgbchannels::Array{Float64,3})
+    # dmw: Can we set this up to return a Gray image instead of an int matrix?
+    # dmw: Check whether the coefficients differ substantially -- we can make it an alternative to the Gray function, like GrayMLB and apply it like Gray.()
     r, g, b = [to_uint8(rgbchannels[:, :, i]) for i in 1:3]
     # Reusing the r array to store the equalized gray image
     r .= to_uint8(0.2989 * r .+ 0.5870 * g .+ 0.1140 * b)
     return r
 end
 
 # dmw: let's figure out the difference between the three versions of this function
-function _process_image_tiles(
+"""
+    conditional_histeq(
+        image,
+        clouds_red,
+        tiles;
+        entropy_threshold::Real=4.0,
+        white_threshold::Real=25.5,
+        white_fraction_threshold::Real=0.4,
+    )
+
+Performs conditional histogram equalization on a true color image.
+
+# Arguments
+- `image`: The true color image to be equalized.
+- `clouds_red`: The land/cloud masked red channel of the false color image.
+- `tiles`: the output from `get_tiles(image)` specifying the tiling to use on the image.
+- `entropy_threshold`: The entropy threshold used to determine if a block should be equalized. Default is 4.0.
+- `white_threshold`: The white threshold used to determine if a pixel should be considered white. Default is 25.5.
+- `white_fraction_threshold`: The white fraction threshold used to determine if a block should be equalized. Default is 0.4.
+
+# Returns
+The equalized true color image.
+
+"""
+function conditional_histeq(
     true_color_image,
     clouds_red,
-    tiles,
-    white_threshold,
-    entropy_threshold,
-    white_fraction_threshold,
+    tiles;
+    entropy_threshold::Real=4.0,
+    white_threshold::Real=25.5,
+    white_fraction_threshold::Real=0.4,
 )
 
     # Apply Perona-Malik diffusion to each channel of true color image 
     # using the default inverse quadratic flux coefficient function
-    true_color_diffused = IceFloeTracker.nonlinear_diffusion(float64.(true_color_image), 0.1, 75, 3)
+    true_color_diffused = IceFloeTracker.nonlinear_diffusion(
+        float64.(true_color_image), 0.1, 75, 3
```

Comment:
I ran the optimization against the matlab diffusion image, and the best parameters are 0.1, 0.1, and 5. Matlab also defaults to exponential, not quadratic. You can import the PeronaMalikDiffusion functor and then replace this line with
```julia
true_color_diffused = IceFloeTracker.nonlinear_diffusion(
        float64.(true_color_image), PeronaMalikDiffusion(0.1, 0.1, 5, "exponential"))
```